plugin/sampling/strategystore/static/strategy_store.go

```diff
@@ -216,35 +219,42 @@ func (h *strategyStore) parseStrategies(strategies *strategies) {
 		newStore.defaultStrategy = h.parseServiceStrategies(strategies.DefaultStrategy)
 	}
 
-	merge := true
-	if newStore.defaultStrategy.OperationSampling == nil ||
-		newStore.defaultStrategy.OperationSampling.PerOperationStrategies == nil {
-		merge = false
-	}
-
 	for _, s := range strategies.ServiceStrategies {
 		newStore.serviceStrategies[s.Service] = h.parseServiceStrategies(s)
 
-		// Merge with the default operation strategies, because only merging with
-		// the default strategy has no effect on service strategies (the default strategy
-		// is not merged with and only used as a fallback).
+		// Config for this service may not have per-operation strategies,
+		// but if the default strategy has them they should still apply.
+
+		if newStore.defaultStrategy.OperationSampling == nil {
+			// Default strategy doens't have them either, nothing to do.
+			continue
+		}
+
 		opS := newStore.serviceStrategies[s.Service].OperationSampling
 		if opS == nil {
-			if newStore.defaultStrategy.OperationSampling == nil ||
+
+			if !h.options.IncludeDefaultOpStrategies &&
```

Comment:
let's not mix up the logic in one place. I suggest copying the original function from main and renaming it xxx_deprecatedI(), so that when the time comes we can just delete it.